data/scripts.js

```diff
@@ -1172,6 +1172,10 @@ let BattleScripts = {
 			if (!pokemon.side.canDynamax) return;
 			if (this.canZMove(pokemon)) return;
 			if (this.canMegaEvo(pokemon)) return;
+			if (pokemon.getItem().zMove) return;
```

Comment:
This is redundant with `this.canZMove(pokemon)`

Reply:
`this.canZMove(pokemon)` does not check after it has used its Z-Move

Reply:
Get rid of `canZMove` basically, you dont need to check if they are still able to use a Z move in general, just if they have a z crystal.

Reply:
This seems wrong to me; we might want to code the restriction in rulesets if there's no engine reason for it.